proxy/gateway/nodejs/lib/control_channel.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+let Channel = require('./channel.js');
+let codec = require('./codec.js');
+
+class ControlChannel extends Channel {
+  constructor() {
+    super();
+    super.on('data', (data) => {
+      let msg = codec.decodeControlMessage(data);
+      if (msg.type === codec.controlMessageTypes.create) {
+        msg = codec.decodeCreateMessage(data);
+        this.emit('create', msg.messageChannelUri, msg.args);
+      }
+      else if (msg.type === codec.controlMessageTypes.start) {
+        this.emit('start');
+      }
+      else if (msg.type === codec.controlMessageTypes.destroy) {
+        this.emit('destroy');
+      }
+    });
+  }
+
+  send(op, arg) {
```

Comment:
Agreed. `send` isn't really used polymorphically (OTOH `connect` and `disconnect` are), so I shouldn't try to make it look like it is. I should probably name the base `send` something else. But I think I'll look the other way, at least for now 😉